back/src/domains/core/file-storage/useCases/UploadFile.ts

```diff
@@ -23,28 +26,26 @@ export class UploadFile extends UseCase<UploadFileInput, string> {
   }
 
   protected async _execute({
-    multerFile,
+    file,
     renameFileToId = true,
-  }: UploadFileInput): Promise<string> {
-    const fileId = renameFileToId
-      ? this.#replaceNameWithUuid(multerFile)
-      : multerFile.originalname;
-
-    const file: StoredFile = {
-      id: fileId,
-      name: multerFile.originalname,
-      encoding: multerFile.encoding,
-      size: multerFile.size,
-      buffer: multerFile.buffer,
-      mimetype: multerFile.mimetype,
+  }: UploadFileInput): Promise<AbsoluteUrl> {
+    const fileToSave: StoredFile = {
+      ...file,
+      id: renameFileToId ? this.#replaceNameWithUuid(file) : file.name,
     };
 
-    await this.#documentGateway.put(file);
-    return this.#documentGateway.getFileUrl(file);
+    const existingFileUrl = await this.#documentGateway.getUrl(fileToSave.id);
+    if (existingFileUrl) throw errors.file.fileAlreadyExist(fileToSave.id);
+
+    await this.#documentGateway.save(fileToSave);
+
+    const savedFile = await this.#documentGateway.getUrl(fileToSave.id);
+    if (!savedFile) throw errors.file.missingFile(fileToSave.id);
+    return savedFile;
   }
 
-  #replaceNameWithUuid(multerFile: Express.Multer.File) {
-    const extension = multerFile.originalname.split(".").at(-1);
+  #replaceNameWithUuid(multerFile: Omit<StoredFile, "id">) {
+    const extension = multerFile.name.split(".").at(-1);
```

Comment:
```suggestion
    const extension = file.name.split(".").at(-1);
```
du coup, plus de multer ?